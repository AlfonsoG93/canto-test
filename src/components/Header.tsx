interface HeaderProps {
	title: string;
}

export default function Header(props: HeaderProps) {
	return (
		<div data-testid={"header-div"}>
			<h1 data-testid={"header-title"}>{props.title}</h1>
		</div>
	);
}