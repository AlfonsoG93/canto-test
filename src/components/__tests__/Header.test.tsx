import React from "react";
import {cleanup, render, screen} from "@testing-library/react";
import Header from "../Header";
import renderer from "react-test-renderer";

const testTitle = "Test Title";


describe("Test Header component", () => {
	afterEach(() => {
		cleanup();
	});
	
	test("Should render component", () => {
		render(<Header title={testTitle}/>);
		const HeaderElem = screen.getByTestId("header-div");
		const HeaderTitle = screen.getByTestId("header-title");
		
		expect(HeaderElem).toBeInTheDocument();
		expect(HeaderTitle).toBeInTheDocument();
		expect(HeaderTitle).toHaveTextContent(testTitle);
		
	});
	
	test("matches snapshot", () => {
		const tree = renderer.create(<Header title={testTitle}/>).toJSON();
		expect(tree).toMatchSnapshot();
		
	});
});