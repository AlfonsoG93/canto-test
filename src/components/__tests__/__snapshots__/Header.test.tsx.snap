// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component matches snapshot 1`] = `
<div
  data-testid="header-div"
>
  <h1
    data-testid="header-title"
  >
    Test Title
  </h1>
</div>
`;
